Source/SwiftLintFramework/Rules/ExplictSelfRule.swift

```diff
@@ -0,0 +1,132 @@
+//
+//  ExplictSelfRule.swift
+//  SwiftLint
+//
+//  Created by Ian Keen on 2016-10-06.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct ExplicitSelfRule: ASTRule, OptInRule, ConfigurationProviderRule {
+    public var configuration = SeverityConfiguration(.Error)
+
+    public init() { }
+
+    public static let description = RuleDescription(
+        identifier: "explicitSelf",
+        name: "Explict Self",
+        description: "Require explicit self for instance members",
+        nonTriggeringExamples: [
+            "class Good1 { let value: Int = 42; var property: Int { return self.value } }",
+            "class Good2 { let value: Int = 42; func function() -> Int { return self.value } }",
+            "class Good3 { let value: Int; init() { self.value = 42 } }",
+        ],
+        triggeringExamples: [
+            "class Bad1 { let value: Int = 42; var property: Int { return value } }",
+            "class Bad2 { let value: Int = 42; func function() -> Int { return value } }",
+            "class Bad3 { let value: Int; init() { value = 42 } }",
```

Comment:
I can think of a few other cases:
- Subclassing: accessing a member from the parent class.
- `struct` and `enum` (because why not)
- `extension`: verify that accessing a member from a function in an extension triggers this too.